File: Python/programmers/etc/level2/다음_큰_숫자.py
def solution(n):
    tmp = n+1
    cnt = bin(n)[2:].count("1")
    while True:
        if cnt == bin(tmp)[2:].count("1"):
            return tmp
        tmp += 1


n = 78
print(solution(n)) # 83

n = 15
print(solution(n)) # 23

File: Java/programmers/etc/level2/다음큰숫자.java
package CodeTest.Java.programmers.etc.level2;

public class 다음큰숫자 {
    public int solution(int n) {
        int answer = n+1;
        int cnt = 0;
        for (char i : Integer.toBinaryString(n).toCharArray()) {
            if (i == '1') cnt++;
        }

        while (true) {
            int tmp = 0;
            for (char i : Integer.toBinaryString(answer).toCharArray()) {
                if (i == '1') tmp++;
            }
            if (cnt == tmp) break;
            answer++;
        }

        return answer;
    }
}
